feat(build): Added Central ``build.py`` Orchestrator for CinderPeak

me-hem opened the pull request:
This PR implements a unified Python-based build orchestrator `build.py` (issue #105) that provides a single entry point for configuring, building, testing, and managing CinderPeak, replacing the need to remember multiple CMake/CTest commands.

## Changes Made
### Core Implementation (`build.py`)
- **Subcommands**: `configure`, `build`, `test`, `clean`, `format`, `check`, `prepare`, `package`
- **Cross-platform support**: Automatic generator detection (Ninja/MSVC/Make) with manual override options
- **Flag passthrough**: `-D key=value` for arbitrary CMake options
- **Data-driven configuration**: `FLAG_MAP` dictionary for easy flag-to-CMake mapping
- **Smart defaults**: Auto-detects build configuration on Windows multi-config generators


### CMakeLists.txt Updates
- Added `SANITIZE` option
- Added `BUILD_COVERAGE` option with coverage flags

## Usage Examples
```bash
# Configure with options
./build.py configure --with-tests --with-examples

# Build with parallel jobs
./build.py build -j 8

# Run tests
./build.py test

# Format/check code
./build.py format --fix
./build.py check tidy

# Setup dev environment
./build.py prepare

# Create source package
./build.py package source --release-version 1.0.0
```

me-hem committed:
Added a Central build.py Orchestrator for CinderPeak and tested on Linux successfully

Commit:
Central Orchestrator modified to make it compatible with Windows and tested successfully

me-hem committed:
fix: changed hook path

SharonIV0x86 commented:
@me-hem The PR is solid but
Here are a few changes can be added.

1. It’d be good to centralize all error handling in a single utility wrapper instead of repeating try/except blocks across functions.
2. The ``run()`` function could be improved by streaming subprocess output live instead of waiting until completion this will help with long builds.
3. In ``configure()``, we can add automatic detection for generators (``Ninja``/``Make``/``MSVC``) to avoid requiring users to specify --generator manually when not needed.
4. grouping all CLI argument definitions and defaults into a dedicated configuration object or dataclass would simplify maintenance and make it easier to update or extend subcommands later.
5. Adding a combined workflow command like ``./build.py all --with-tests`` would be easier to and faster.

me-hem commented:
@SharonIV0x86 I will look into it.

me-hem committed:
Improved build.py with streaming output, generators auto-detection, and unified workflow

me-hem commented:
@SharonIV0x86 I have improved the build orchestrator `build.py` with streaming output, generator auto-detection and unified workflow command.

### Key Changes:
- Added live subprocess output streaming instead of buffering
- Added auto-detection for best available generator (Ninja/Make/MSVC)
- Added Config class to centralize all default configuration values
- Added `@handle_build_errors` decorator to all command functions for consistent error handling
- Added `all` command for unified configure+build+test workflow in one step
- Modified functions to use Config class defaults instead of hardcoded values

### Usage Example: 
```
./build.py all --with-tests --with-examples --sanitize -j 8 --config Debug
```

SharonIV0x86 reviewed (approved):
Lgtm.

SharonIV0x86 commented:
@me-hem Thank you for contributing, your PR has been merged.
